### ISSUE TITLE
NiFi fails to execute generate.sh due to permission error

### ISSUE THREAD

**Issue by ijokarumawak** [hortonworks/data-tutorials#180]
Please Fill in the Following Information before submitting your Issue:

**Tutorial Name**:  [Realtime Event Processing in Hadoop with NiFi, Kafka and Storm](http://hortonworks.com/hadoop-tutorial/realtime-event-processing-nifi-kafka-storm/#section_3)
**HDP Version are you Running**: 2.5
**Applicable HDP Components**: 

**Link to tutorial**(Hortonworks or GitHub website):
https://github.com/hortonworks/tutorials/blob/hdp-2.5/tutorials/hortonworks/realtime-event-processing-with-hdf/rtep-1.md
### Detailed Description of Issue:

In the tutorial, NiFi is configure to call generate.sh using ExecuteProcess processor. However, since NiFi process is launched by 'nifi' user and the generate.sh is under /root directory, it can't be executed from NiFi with permission error.

Users are not able to complete the tutorial because of this, several users have asked questions in [HCC](https://community.hortonworks.com/questions/61844/nifi-executeprocess-does-not-generate-realtime-dat.html#answer-62058).
### Proposed Fix:

Update the tutorial to clone the Git project on some different directory, such as /opt so that 'nifi' user can refer it.

**Comment by orendain** [comment]
Good catch, should definitely get that fixed - common issue to get snagged on with NiFi.

Author @james94 What do you think?

**Comment by james94** [comment]
@ijokarumawak thank you for the suggestion and great idea. I will work on making that change soon.

**Comment by TheBellman** [comment]
It would probably be cleanest to update the overall instructions to ensure that the repository is checked out as `nifi` in `/home/nifi`, and the Maven build executed there.

**Comment by TheBellman** [comment]
I would also suggest that the _PutFile_ processors in that tutorial are configured to point somewhere other than `/root/nifi_output`, as that will be another potential source of permissions problems - a previous tutorial should have executed `./auto_scripts/init_nifi_in_src_data.sh` to create `/tmp/nifi/input` with very broad permissions, so using `/tmp/nifi/output` may be a better option.

**Comment by james94** [comment]
Hey @ijokarumawak @TheBellman 
I modified the tutorial based on your suggestions. I tested on the NiFi and Kafka portion in the tutorial series with HDP 2.6.4 that runs in HDF 3.0.2 and was able to see messages being persisted into the Kafka topic "truck_event" from the NiFi dataflow. I am awaiting one of my teammates to review my pull request for this tutorial. After the tutorial is modified, I will send you both an update.
